tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenTestFactory.cs

```diff
@@ -27,67 +28,77 @@ public class JsonDrivenTestFactory
         private readonly string _databaseName;
         private readonly string _collectionName;
         private readonly Dictionary<string, object> _objectMap;
+        private readonly ITestRunner _testRunner;
 
         // public constructors
         public JsonDrivenTestFactory(IMongoClient client, string databaseName, string collectionName, Dictionary<string, object> objectMap)
+            : this(null, client, databaseName, collectionName, objectMap)
+        {
+        }
+
+        public JsonDrivenTestFactory(ITestRunner testRunner, IMongoClient client, string databaseName, string collectionName, Dictionary<string, object> objectMap)
         {
             _client = client;
             _databaseName = databaseName;
             _collectionName = collectionName;
             _objectMap = objectMap;
+            _testRunner = testRunner;
         }
 
         // public methods
         public JsonDrivenTest CreateTest(string receiver, string name)
         {
-            if (receiver.StartsWith("session"))
-            {
-                switch (name)
-                {
-                    case "abortTransaction": return new JsonDrivenAbortTransactionTest(_objectMap);
-                    case "commitTransaction": return new JsonDrivenCommitTransactionTest(_objectMap);
-                    case "startTransaction": return new JsonDrivenStartTransactionTest(_objectMap);
-                    case "withTransaction": return new JsonDrivenWithTransactionTest(this, _objectMap);
-                    default: throw new FormatException($"Invalid method name: \"{name}\".");
-                }
-            }
-
-            var database = _client.GetDatabase(_databaseName);
-            if (receiver == "database")
+            switch (receiver)
             {
-                switch (name)
-                {
-                    case "runCommand": return new JsonDrivenRunCommandTest(database, _objectMap);
-                    default: throw new FormatException($"Invalid method name: \"{name}\".");
-                }
+                case "testRunner":
+                    switch (name)
+                    {
+                        case "targetedFailPoint": return new JsonDrivenTargetedFailPointTest(_testRunner.ConfigureFailPoint, _testRunner.ConfigureFailpointAsync, _objectMap);
+                        case "assertSessionPinned": return new JsonDrivenAssertSessionPinnedTest(_objectMap);
+                        case "assertSessionUnpinned": return new JsonDrivenAssertSessionUnpinnedTest(_objectMap);
```

Comment:
Pass the _testRunner to all three tests.

This is analogous to passing the database or collection to database or collection tests.